include/envoy/network/listener.h

```diff
@@ -113,10 +113,10 @@ class ListenerCallbacks {
  * TODO(conqerAtapple): Maybe this belongs inside the UdpListenerCallbacks
  * class.
  */
-struct UdpData {
-  Address::InstanceConstSharedPtr local_address_;
-  Address::InstanceConstSharedPtr peer_address_; // TODO(conquerAtapple): Fix ownership semantics.
+struct UdpRecvData {
+  Address::InstanceConstSharedPtr peer_address_;
```

Comment:
Is self_address_ the local_address_? If so, I have added it on the interface of the udp listener. Is that sufficient rather than pumping that info into every message?

Reply:
I assume you meant to make them the same thing. But I'm not sure if a unified local_address_ for listener make sense or not in presence of virtual IP. Please see my comment in `local_address`.

No matter if self_address_ here should be a reference to UdpListener::local_address() or a per-packet value. UdpRecvData is what got passed into ListenerCallbacks::onData() so I think embed self_address_ in it is a good way of passing  this piece information to the UdpListenerCallbacks instance.

Reply:
okay, i can alter that.
do you prefer the new UdpRecvData or the older UdpData?

Reply:
Either way is fine to me. Maybe UdpRecvData is more intuitive?

Reply:
ack

Reply:
ack